### PR TITLE
[C#] Extract concepts of v2 nucleotide-count exercise

### LINKED ISSUE #260: [C#] Extract Concepts from v2 exercise: nucleotide-count
Extract the list of concepts required to solve the v2 C# exercise: `nucleotide-count`. For each concept, describe why it is required to solve the exercise. Keep in mind that an exercise can have multiple approaches. The goal of this task is to help determine the total list of concepts that make up `nucleotide-count`.

The list of concepts should be stored in the following document: `languages/csharp/reference/exercise-concepts/nucleotide-count.md`. [Use this link to create a PR to add this document][pr-link].

[pr-link]: ../new/master?filename=languages/csharp/reference/exercise-concepts/nucleotide-count.md

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes #260

@robkeim @shaynuhcon For your inspiration, this is my first extraction of concepts.

### CODE REVIEW

File: languages/csharp/reference/exercise-concepts/nucleotide-count.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Concepts of nucleotide-count
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/nucleotide-count.md]: Can you add a link to the v2 implementation right after the header? I'm going to do the same with my exercises as well.

File: languages/csharp/reference/exercise-concepts/nucleotide-count.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## General
+
+- functions: used as the main entry point for the exercise
+- function arguments: input strand is passed as an arguments
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/nucleotide-count.md]:
```suggestion
- function arguments: input strand is passed as an argument
```

File: languages/csharp/reference/exercise-concepts/nucleotide-count.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+- object initializers: initializing dictionary through object initializer
+
+## Approach: using TryGetValue
+
+- out parameters: capturing the value as an `out` parameter
+
+## Notes
+
+- Having the exercise return an `IDictionary<char, int>` means that the users will have to know about interfaces too. This was not the goal of the exercise (teaching about dictionaries is), so it's interesting to see that this was missed. Good thing to keep in mind when develop Concept Exercises.
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/nucleotide-count.md]:
```suggestion
- Having the exercise return an `IDictionary<char, int>` means that the users will have to know about interfaces too. This was not the goal of the exercise (teaching about dictionaries is), so it's interesting to see that this was missed. Good thing to keep in mind when developing Concept Exercises.
```

**Comment by robkeim** [comment]
@ErikSchierboom should I make any effort in my PR to align with the terminology that you used in yours or are you going to do that in the "aggregation" phase once we have all of the exercises broken down into concepts?

File: languages/csharp/reference/exercise-concepts/nucleotide-count.md (hunk 1 above)
Comment (ErikSchierboom) [reply languages/csharp/reference/exercise-concepts/nucleotide-count.md]: Done!

**Comment by ErikSchierboom** [comment]
> should I make any effort in my PR to align with the terminology that you used in yours or are you going to do that in the "aggregation" phase once we have all of the exercises broken down into concepts?

@robkeim  We'll do that in the aggregation phase.

**Pull request merged**